driver: input: sbus: Add connection key code

yunsmall opened the pull request:
The disconnection and connection of the remote control is a very important event. If this information cannot be obtained at the application layer, it is easy to cause unexpected dangerous situations for the operator. Therefore, the function of obtaining this information is added.

dsseng commented on drivers/input/input_sbus.c:
```diff
@@ -129,6 +134,15 @@ static void input_sbus_input_report_thread(const struct device *dev, void *dummy
 	bool connected_reported = false;
 
+	/* Before officially starting the report, let's first report a disconnection. */
+	if(cfg->connection_code != INPUT_KEY_RESERVED){
```
Please adhere to the code style as in other parts of the file

```suggestion
	if (cfg->connection_code != INPUT_KEY_RESERVED) {
```

dsseng reviewed (commented):
thanks, could you please format the new code to match the style of existing one (and therefore the guidelines)?

PetervdPerk-NXP commented:
Thanks for the PR. I had actually been looking for a way to expose the RC connection status but hadn't gotten around to it yet. While we can certainly leverage event codes for this, ideally there should be a standardized way for all types of input devices to indicate availability.

@fabiobaltieri what's your take? Is this the best approach for indicating connection status, or have you seen other patterns used elsewhere?

fabiobaltieri commented:
> @fabiobaltieri what's your take? Is this the best approach for indicating connection status, or have you seen other patterns used elsewhere?

Not really, I think it may be fair to make this generate a `INPUT_EV_DEVICE` event and go with it for now, if we see more drivers needing it we can try to figure a convention but for now this would unblock any extra signal the driver may want to send.

So just this but using `input_report(dev, INPUT_EV_DEVICE, ...`, the application can obviously only rely on this if it knows what devices is sending it.

PetervdPerk-NXP commented:
@yunsmall 

Could you use `INPUT_EV_DEVICE` instead of `INPUT_EV_KEY` this removed the need as well for the dts configuration.
You can then expose the `INPUT_EV_DEVICE` through the `sbus.h` header.

yunsmall commented:
@PetervdPerk-NXP I have no clue why my CLion decided to use my own .clang-format file, but it totally messed up the formatting. 😅

yunsmall committed:
drivers: input: sbus: Add connection key code

The disconnection and connection of the remote control
is a very important event. If this information cannot
be obtained at the application layer, it is easy to
cause unexpected dangerous situations for the operator.
Therefore, the function of obtaining this information
is added.

Signed-off-by: Yunjie Ye <yun_small@163.com>
